src/odb/src/upf/upf.cpp

```diff
@@ -217,4 +221,631 @@ bool set_isolation(utl::Logger* logger,
+                   "data or enable terms.",
+                   iso->getName());
+      continue;
+    }
+
+    // Determine if an inverter is needed for the control pin & output
+    auto func = out_lib_port->function();
+    int inv_needed = check_isolation_match(
+        func, enable_lib_port, isolation_sense, isolation_clamp_val);
+
+    // find the smallest possible inverter
+    odb::dbMaster* inverter_m = nullptr;
+    odb::dbMTerm *input_m = nullptr, *output_m = nullptr;
+
+    if (inv_needed > 0
+        && find_smallest_inverter(logger, block, inverter_m, input_m, output_m)
+               == false) {
+      logger->warn(utl::ODB, 130482, "can't find any inverters");
+      continue;
+    }
+
+    // Iterate through all pd's instances and determine if any signal needs
+    // isolation If one is found, create dbInst from the isolation cell +
+    // connect control signal of isolation to dbInst
+    // connect previous net to data input
+    // connect output from dbInst to
+
+    odb::dbGroup* curr_group = domain->getGroup();
+    auto insts = curr_group->getInsts();
+
+    for (auto&& inst : insts) {
+      auto iterms = inst->getITerms();
+      for (auto&& iterm : iterms) {
+        if (iterm->getIoType() != odb::dbIoType::OUTPUT)
+          continue;
+
+        auto net = iterm->getNet();
+
+        if (!net) {
+          continue;
+        }
+
+        auto connectedIterms = net->getITerms();
+
+        if (connectedIterms.size() < 2)
+          continue;
+
+        // Find ITERMS that belong to instances outside of this power domain
+        std::vector<odb::dbITerm*> external_iterms;
+        for (auto&& connectedIterm : connectedIterms) {
+          auto connectedInst = connectedIterm->getInst();
+
+          if (!connectedInst->getGroup()
+              || connectedInst->getGroup()->getName()
+                     != curr_group->getName()) {
+            external_iterms.push_back(connectedIterm);
+          }
+        }
+
+        if (external_iterms.size() < 1)
+          continue;
+        // Create isolaton cell if there exists connected instances outside of
+        // the power domain Add it to its specified location
+        std::string inst_name = inst->getName() + "_isolation_cell";
+        auto isolation_inst
+            = odb::dbInst::create(block, smallest_iso_m, inst_name.c_str());
+
+        if (iso->getLocation() == "self") {
+          curr_group->addInst(isolation_inst);
+        } else if (iso->getLocation() == "parent") {
+          auto ppd = domain->getParent();
+          if (ppd && ppd->getGroup()) {  // if the parent domain is the top
+                                         // domain, don't add to any group
+            ppd->getGroup()->addInst(isolation_inst);
+          }
+        } else {
+          logger->warn(utl::ODB,
+                       15022,
+                       "Isolation %s has location %s, but only self|parent "
```

Comment:
Isn't there an option to put it in the sink's domain (possibly more than one)?

Reply:
there's a fanout option which I also added support for this commit